src/common.js

```diff
@@ -4,3 +4,9 @@ export const settings = new Gio.Settings({
   schema_id: "re.sonny.Junction",
   path: "/re/sonny/Junction/",
 });
+
+export class CommonSettings {
+
+   static openedWindows = new Set();
```

Comment:
Approach should be to detect opened windows and act on this information. For example window could be killed by third party program, then this logic would not work.